core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationDirective.js

```diff
@@ -0,0 +1,131 @@
+
+          $scope.isActive = function(tabId) {
+            return ($scope.ACTIVATED_TAB_ID === tabId);
+          };
+
+          $scope.onTabClick = function(tabId) {
+            $scope.ACTIVATED_TAB_ID = tabId;
+            $scope.activeHintIndex = null;
+            $scope.activeAnswerGroupIndex = null;
+          };
+
+          $scope.isDisabled = function(tabId) {
+            if (tabId === $scope.TAB_ID_FEEDBACK) {
+              if (!$scope.stateDefaultOutcome || !$scope.stateInteractionId) {
+                return true;
+              } else {
+                return false;
+              }
+            } else if (tabId === $scope.TAB_ID_HINTS) {
+              if ($scope.stateHints.length <= 0) {
+                return true;
+              } else {
+                return false;
+              }
+            } else {
+              if (!$scope.stateSolution) {
+                return true;
+              } else {
+                return false;
+              }
+            }
+          };
+
+          $scope.changeActiveHintIndex = function(newIndex) {
+            // If the current hint is being clicked on again, close it.
+            if (newIndex === $scope.activeHintIndex) {
+              $scope.activeHintIndex = null;
+            } else {
+              $scope.activeHintIndex = newIndex;
+            }
+          };
+
+          $scope.changeActiveAnswerGroupIndex = function(newIndex) {
+            // If the current answer group is being clicked on again, close it.
+            if (newIndex === $scope.activeAnswerGroupIndex) {
+              $scope.activeAnswerGroupIndex = null;
+            } else {
+              $scope.activeAnswerGroupIndex = newIndex;
+            }
+          };
+
+
+          $scope.getHtmlSummary = function(subtitledHtml) {
+            var htmlAsPlainText = $filter(
+              'formatRtePreview')(subtitledHtml.getHtml());
+            return htmlAsPlainText;
+          };
+
+          $scope.$on('refreshStateTranslation', function() {
+            $scope.initStateTranslation();
+          });
+          $scope.initStateTranslation = function() {
+            $scope.ACTIVATED_TAB_ID = $scope.TAB_ID_CONTENT;
+
+            var stateName = EditorStateService.getActiveStateName();
+            $scope.stateContent = ExplorationStatesService
+              .getStateContentMemento(stateName);
+            $scope.stateSolution = ExplorationStatesService
+              .getSolutionMemento(stateName);
+            $scope.stateHints = ExplorationStatesService
+              .getHintsMemento(stateName);
+            $scope.stateAnswerGroups = ExplorationStatesService
+              .getInteractionAnswerGroupsMemento(stateName);
+            $scope.stateDefaultOutcome = ExplorationStatesService
+              .getInteractionDefaultOutcomeMemento(stateName);
+            $scope.stateInteractionId = ExplorationStatesService
+              .getInteractionIdMemento(stateName);
```

Comment:
So I noticed one slightly strange behavior, and I think it has to do with these mementos. If I am on the translation tab, and the refresh the page, all of the material disappears. But if I go to main editor tab first, and then come back the material is initialized. Should this be happening?

Reply:
Done!